Wikipedia/Code/WMFAppViewController+Extensions.swift

```diff
@@ -231,9 +231,33 @@ extension WMFAppViewController: WKWatchlistDelegate {
             navigate(to: siteURL.wmf_URL(withPath: "/wiki/User_talk:\(username)", isMobile: true))
         case .userContributions:
             navigate(to: siteURL.wmf_URL(withPath: "/wiki/Special:Contributions/\(username)", isMobile: true))
-        case .thank:
-            // TODO
-            break
+        case .thank(let revisionID):
```

Comment:
@staykids Do you think we could lean on [this protocol](https://github.com/wikimedia/wikipedia-ios/blob/main/Wikipedia/Code/ThanksGiving.swift) that diffs use? It looks like it's already set up with a login and onboarding prompt if needed, plus some logging. I don't think there's a reason it needs to be limited to `ViewController` classes, but I could be wrong.

Reply:
👍🏽 sounds good – thanks for bringing it up to begin with!